File: cpp/alerts/storage_moved.cpp
#include "./storage_moved.h"

namespace ltrs {

rust::Str storage_moved_alert_get_storage_path(lt::storage_moved_alert* alert) {
    return alert->storage_path();
}

rust::Str storage_moved_alert_get_old_path(lt::storage_moved_alert* alert) {
    return alert->old_path();
}

} // namespace ltrs

File: src/ffi/alerts/storage_moved.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type storage_moved_alert = crate::ffi::ffi::storage_moved_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/storage_moved.h");
        unsafe fn storage_moved_alert_get_storage_path<'a>(
            alert: *mut storage_moved_alert,
        ) -> &'a str;
        unsafe fn storage_moved_alert_get_old_path<'a>(alert: *mut storage_moved_alert) -> &'a str;
    }
}
